ironfish/src/rpc/routes/wallet/getAsset.ts

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { ASSET_ID_LENGTH } from '@ironfish/rust-nodejs'
+import * as yup from 'yup'
+import { Assert } from '../../../assert'
+import { CurrencyUtils } from '../../../utils'
+import { NotFoundError, ValidationError } from '../../adapters'
+import { ApiNamespace, routes } from '../router'
+import { getAccount } from './utils'
+
+export type GetWalletAssetRequest = {
+  account?: string
+  confirmations?: number
+  id: string
+}
+
+export type GetWalletAssetResponse = {
+  createdTransactionHash: string
+  creator: string
+  id: string
+  metadata: string
+  name: string
+  nonce: number
+  status: string
+  // Populated for assets the account owns
+  supply: string | null
+  // Populated once the asset has been added to the main chain
+  blockHash: string | null
+  sequence: number | null
```

Comment:
There are a few differences between this response and the one from `getAssets`, should we make them the same?

Reply:
Yep, good catch. Updated